Session/Home/HomeViewModel.swift

```diff
@@ -578,16 +617,44 @@ public class HomeViewModel: NavigatableStateHolder {
         dependencies[defaults: .standard, key: .rateAppRetryAttemptCount] = 0
         
         if let scene = UIApplication.shared.connectedScenes.first(where: { $0.activationState == .foregroundActive }) as? UIWindowScene {
-            SKStoreReviewController.requestReview(in: scene)
+            NotificationCenter.default.addObserver(
+                self,
+                selector: #selector(self.systemHasPresentedNewWindow(notification:)),
+                name: UIWindow.didBecomeVisibleNotification, object: nil
+            )
+            
+            if !dependencies[feature: .simulateAppReviewLimit] {
+                SKStoreReviewController.requestReview(in: scene)
+            }
+
+            // Added 2 sec delay to give time for requet review to proc
+            DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) { [self] in
+                NotificationCenter.default.removeObserver(self, name: UIWindow.didBecomeVisibleNotification, object: nil)
```

Comment:
While it doesn't currently happen, we should probably add `NotificationCenter.default.removeObserver(self)` to the `HomeViewModel.deinit` just in case it gets destroyed during this delay